File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    
    cnt = 0
    start = 0
    end = len(people) - 1
    people.sort()
    
    while end > start:
        if people[start] + people[end] <= limit:
            people[start] = -1
            people[end] = -1
            start += 1
            end -= 1
            cnt += 1
        else:
            end -= 1
    
    for i in people:
        if i != -1:
            cnt += 1
    
    return cnt
        

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        
        ArrayList<Integer> tmp = new ArrayList<>();
        for(int i : people){
            tmp.add(i);
        }
        tmp.sort((o1,o2) -> {return o1 - o2;});
        
        Deque<Integer> lis = new ArrayDeque<>();
        for(int i : tmp){
            lis.add(i);
        }
        
        int ans = 0;
        
        while(lis.size() > 0){
            int capa = limit - lis.removeFirst();
            while(lis.size() > 0){
                int now = lis.removeLast();
                if(capa >= now){
                    break;
                }
                else{
                    ans ++;
                }
            }
            ans ++;
        }
        return ans;
    }
}